Apps/pharcsight/arcsight_connector.py

```diff
@@ -0,0 +1,837 @@
+        if (phantom.is_fail(ret_val)):
+            return action_result.get_status()
+
+        return phantom.APP_SUCCESS
+
+    def _get_version(self, action_result):
+
+        endpoint = "{0}/getESMVersion".format(ARCSIGHT_CASESERVICE_ENDPOINT)
+
+        params = {'authToken': self._auth_token}
+
+        ret_val, resp = self._make_rest_call(endpoint, action_result, params)
+
+        if (phantom.is_fail(ret_val)):
+            return (phantom.APP_ERROR, resp)
+
+        self.debug_print(resp)
+
+        return (phantom.APP_SUCCESS, resp)
+
+    def _parse_error(self, response):
+
+        try:
+            soup = BeautifulSoup(response.text, "html.parser")
+            pres = soup.findAll('pre')
+            error_text = '\r\n'.join([str(x) for x in pres])
+        except:
+            error_text = "Cannot parse error details"
+
+        # Try to parse some more
+        try:
+            error_text = str(pres[0]).split('\n')[0].replace('<pre>', '')
+        except:
+            pass
+
+        message = ARCSIGHT_ERR_FROM_SERVER.format(status=response.status_code, message=error_text)
+
+        return message
+
+    def _make_rest_call(self, endpoint, action_result, params=None, data=None, json=None, headers=None, method="get"):
+
+        config = self.get_config()
+
+        request_func = getattr(requests, method)
+
+        if (not request_func):
+            action_result.set_status(phantom.APP_ERROR, ARCSIGHT_ERR_API_UNSUPPORTED_METHOD, method=method)
+
+        # self.save_progress("Connecting to {0}...".format(self._base_url))
+        url = self._base_url + endpoint
+
+        self.debug_print("Making REST Call {0} on {1}".format(method.upper(), url))
+
+        _headers = {'Accept': 'application/json'}
+
+        if (headers):
+            _headers.update(headers)
+
+        try:
+            response = request_func(url, params=params, data=data, json=json, headers=_headers, verify=config[phantom.APP_JSON_VERIFY])
+        except Exception as e:
+            self.debug_print("REST call Failed: ", e)
+            return (action_result.set_status(phantom.APP_ERROR, ARCSIGHT_ERR_SERVER_CONNECTION, e), None)
+
+        # self.debug_print('REST url: {0}'.format(response.url))
+
+        if (response.status_code != requests.codes.ok):  # pylint: disable=E1101
+            message = self._parse_error(response)
+            self.debug_print(message)
+            return (action_result.set_status(phantom.APP_ERROR, message), None)
+
+        reply = response.text
+
+        action_result.add_debug_data(reply)
+
+        try:
+            response_dict = response.json()
+        except Exception as e:
+            self.save_progress(ARCSIGHT_ERR_UNABLE_TO_PARSE_REPLY)
+            return (action_result.set_status(phantom.APP_ERROR, ARCSIGHT_ERR_UNABLE_TO_PARSE_REPLY, e), None)
+
+        return (phantom.APP_SUCCESS, response_dict)
+
+    def _get_case_events(self, event_ids, action_result):
+
+        if (type(event_ids) is not list):
```

Comment:
```suggestion
        if not isinstance(event_ids, list):
```

ditto for other cases of `type(...) is ...` or `type(...) is not ...`

Reply:
@rgil-splunk 
If you see above comment by Khushank, Matt asked us not cover any efforts for Python2to3 conversion or any DEV checks, just convert it to Uncertified and release it.
Do you still want us to perform these changes?